src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,377 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
+        }
+    } else if (!D_80AE81AC) {
+        D_80AE81AC = 1;
+    }
+}
+
+EnRlActionFunc sActionFuncs[] = { func_80AE7798, func_80AE77B8, func_80AE77F8, func_80AE7838,
+                                  func_80AE7C64, func_80AE7C94, func_80AE7CE8, func_80AE7D40 };
+EnRlDrawFunc sDrawFuncs[] = { func_80AE7FD0, func_80AE7FDC, func_80AE7D94 };
+
+const ActorInit En_Rl_InitVars = { ACTOR_EN_RL,
+                                   ACTORTYPE_NPC,
+                                   FLAGS,
+                                   OBJECT_RL,
+                                   sizeof(EnRl),
+                                   (ActorFunc)EnRl_Init,
+                                   (ActorFunc)EnRl_Destroy,
+                                   (ActorFunc)EnRl_Update,
+                                   (ActorFunc)EnRl_Draw };
+
+void func_80AE744C(EnRl* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 5);
+}
+
+s32 func_80AE7494(EnRl* this) {
+    return SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+s32 func_80AE74B4(EnRl* this, GlobalContext* globalCtx, u16 arg2, s32 arg3) {
+    CsCmdActorAction* csCmdActorAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        csCmdActorAction = globalCtx->csCtx.npcActions[arg3];
+        if (csCmdActorAction != NULL && csCmdActorAction->action == arg2) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+s32 func_80AE74FC(EnRl* this, GlobalContext* globalCtx, u16 arg2, s32 arg3) {
+    CsCmdActorAction* csCmdActorAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        csCmdActorAction = globalCtx->csCtx.npcActions[arg3];
+        if (csCmdActorAction != NULL && csCmdActorAction->action != arg2) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void func_80AE7544(EnRl* this, GlobalContext* globalCtx) {
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06007B38, &D_06000A3C, 0, 0, 0);
+}
+
+void func_80AE7590(EnRl* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player;
+    Vec3f pos;
+    s16 sceneNum = globalCtx->sceneNum;
+
+    if (gSaveContext.sceneSetupIndex == 4 && sceneNum == SCENE_KENJYANOMA && globalCtx->csCtx.state != 0 &&
+        globalCtx->csCtx.npcActions[6] != NULL && globalCtx->csCtx.npcActions[6]->action == 2 && !this->lightMedallionGiven) {
+        player = PLAYER;
+        pos.x = player->actor.posRot.pos.x;
+        pos.y = player->actor.posRot.pos.y + 80.0f;
+        pos.z = player->actor.posRot.pos.z;
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_EFFECT, pos.x, pos.y, pos.z, 0, 0, 0, 0xE);
+        Item_Give(globalCtx, ITEM_MEDALLION_LIGHT);
+        this->lightMedallionGiven = 1;
+    }
+}
+
+void func_80AE7668(EnRl* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 temp;
+
+    this->drawConfig = 1;
+    this->action = 1;
+    temp = this->actor.posRot.rot.y + 0x8000;
+    player->actor.shape.rot.y = temp;
+    player->actor.posRot.rot.y = temp;
```

Comment:
You can probably do this to get rid of the temp
```suggestion
    player->actor.posRot.rot.y = player->actor.shape.rot.y = this->actor.posRot.rot.y + 0x8000;
```